### DISCUSSION TITLE
Dependency injection

### DISCUSSION THREAD

**Discussion by oliverbj** [livewire/livewire#1116] [4. General]
Hi there!

I have a model, called `Container.php`, that I am trying to pass to a Livewire component in my controllers `index.blade.php`page. My controllers method simply looks like this:

```php
//Controller.php
public function index(Container $container)
{
     return view('containers.train.index', compact('container'));
}
```

```php
//index.blade.php
@extends('layouts.app')

@section('content')
    @livewire('containers.train.text-classification', ['container' => $container])
@endsection
```

Inside my `TextClassification.php` livewire component, it looks like this:

```php
use App\Container;
class TextClassification extends Component
{

    public $container;

    public function mount(Container $container)
    {
        $this->container = $container;
    }

    public function render()
    {
        return view('livewire.containers.train.text-classification');
    }
}
```

Is this the right way of initializing a model from the route model binding from the controller?

**Comment by nuernbergerA** [comment]
That's the only way to pass a variable to a Livewire `Component`.

I'm not sure what you mean with `route model binding`, because you just pass a variable and livewire has no clue if its set by a route.

**Reply by oliverbj** [reply]
Ah I see. Maybe it was in precious Livewire versions, where you could just pass it like this:

```
@livewire(‘component’, $variable)
```

I remember back then, that we could not pass a model directly to Livewire while having it set to `public`, but it seems to work now.

Just wondering if I am doing it correctly.

Thanks!
